Factor out isDominatedBy

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet
* #56642 Only optimize while change has been made
* #56438 [JIT] integer value refinement
* **#56437 Factor out isDominatedBy**
* #55978 Add Value * == Value * peephole
* #55977 [JIT] [Easy] Factor out non tensor peephole
* #55926 Add list len refinement
* #55925 Add handling of symbolic shapes
* #54809 Initial Symbolic Shape Analysis

Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

Commit:
Factor out isDominatedBy

[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"




[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"




[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"




[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"




[ghstack-poisoned]

penguinwu commented on torch/csrc/jit/ir/ir.h:
```diff
@@ -264,8 +264,24 @@ struct Value {
+  // Replaces all uses of this value with 'newValue' that are dominated by
+  // 'node'. replaceAllUsesAfterNodeWith does not check domination, so in the
+  // following example:
+  // x = op(...) if cond:
```
=> 
```
x = op(...)
if cond:
```

eellison replied:
clang format 😡

penguinwu commented on torch/csrc/jit/ir/ir.h:
```diff
@@ -264,8 +264,24 @@ struct Value {
   TORCH_API void replaceAllUsesAfterNodeWith(const Node* node, Value* newValue);
 
+  // Replaces all uses of this value with 'newValue' that are dominated by
+  // 'node'. replaceAllUsesAfterNodeWith does not check domination, so in the
```
The comment shows why using `replaceAllUsesAfterNodeWith` may be wrong. Perhaps it is more direct to just show how `replaceAllUsesDominatedByNodeWith` would transform the example (so that people don't have to understand what `repalceAllUsesAfterNodeWith` is).

penguinwu reviewed (approved):
LGTM, just some minor commments

Commit:
Update on "Factor out isDominatedBy"




[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

Commit:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).

eellison committed:
Update on "Factor out isDominatedBy"


Differential Revision: [D27924240](https://our.internmc.facebook.com/intern/diff/D27924240)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27924240).
